tests/unit/test_bids_utils.py

```diff
@@ -81,3 +86,66 @@ def test_bidsmap_scans_run_plus():
     scans = [("1", "run+"), ("2", "run+"), ("3", "run+")]
 
     assert bidsmap_scans(scans) == [("1", "run-01"), ("2", "run-02"), ("3", "run-03")]
+
+
+def test_bidsify_dicom_headers(mocker):
+    """Test bidsify_dicom_headers without ProtocolName field"""
+    dataset = mocker.MagicMock()
+    dataset.__contains__.return_value = False
+
+    dcmread = mocker.patch("pydicom.dcmread")
+    dcmread.return_value = dataset
+
+    bidsify_dicom_headers("filename", "foo")
+
+    assert dataset.data_element.called == False
+
+
+def test_bidsify_dicom_headers_with_protocol_name(mocker):
+    """Test bidsify_dicom_headers with ProtocolName match"""
+    series_description = "foo"
+
+    dataset = mocker.MagicMock()
+    dataset.__contains__.return_value = True
+    dataset.data_element.return_value = mocker.Mock(value=series_description)
```

Comment:
4. In this test, we're setting `__contains__` return value to `True` meaning that `"ProtocolName"` is in `dataset`. We get to continue to the second if statement in that function. The second if statement looks like this:

```python
if dataset.data_element("ProtocolName").value != series_description:
    pass
```

Here we're calling `data_element()` on `dataset` and getting `.value` on whatever `data_element()` returns. We need to add a `data_element` method to our mock, and we need to change its return value to something with the attribute `.value`. The easiest way for me to do that is with another mock. `Mock(value=series_description)` create a new mock with a `value` attribute set to `series_description`. When we hit that if statement `dataset.data_element("ProtocolName").value` evaluates to whatever was in `series_description` when we created the new mock